### PR TITLE
jsoncpp 1.6.5

### PR DESCRIPTION
**Pull request by theopolis**
### All Submissions:

- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

### Note

The `1.x.y` version requires c++11. As long as there're no issues with downstream builds, this should be preferred. Some of the std containers in the `0.x.y` branch are deprecated and really shouldn't be used with modern runtimes.

### CODE REVIEW

**Comment by dunn** [comment]
Looks like `libjson-rpc-cpp` needs to be rebuilt against this new version; add `revision 1` below its `head` line. Thanks!

**Comment by bfontaine** [comment]
@theopolis Please do it in a separate commit. You can use e.g. `libjson-rpc-cpp: bump the revision` for the commit message.

**Comment by theopolis** [comment]
Ok @bfontaine, I took away the revision bump on `libjson-rpc-cpp`, that means the BrewBot tests will fails this PR when rebuilding the forward dependencies: the required PR #50087 must land once this has been merged.

**Comment by dunn** [comment]
It can be in the same PR, just in a separate commit.

**Comment by DomT4** [comment]
> The 1.x.y version requires c++11

Does the formula need a `needs :cxx11` and `ENV.cxx11` then? Not completely sure. Will defer to another maintainer.

**Comment by theopolis** [comment]
There shouldn't be any harm adding them, right?

File: Library/Formula/jsoncpp.rb [hunk 1]
```diff
@@ -15,11 +15,17 @@ class Jsoncpp < Formula
 
   option :universal
 
+  needs :cxx11
+
   depends_on "cmake" => :build
 
   def install
+    ENV.cxx11
+    ENV.append "CXXFLAGS", "-std=c++11"
```
Comment (DomT4) [inline Library/Formula/jsoncpp.rb]: `ENV.cxx11` should be doing some of this intelligently for you. I'm surprised you need either of those lines below that?

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/be8bfbd238e056ba9a825537eb731a3707d1f6d9 & https://github.com/Homebrew/homebrew/commit/040135e92e2c43f242fd0f9318b03a71adb4c2c1. Thank you once again @theopolis! :smiley_cat:

**Pull request closed without merging**